### PR TITLE
second lesson tasks from lab and homework

### PR DESCRIPTION
**Pull request by ezjust**
please confirm that homework done in proper way)

### CODE REVIEW

File: second_lesson_home_tasks.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import math
+import random
+
+'''Home tasks, 1.5b'''
+#1
```
Comment (itea-artem) [inline second_lesson_home_tasks.py]: why this code in comments?

File: second_lesson_home_tasks.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+#     print("factorial is: ", 1)
+# else:
+#     print("factorial is: ", math.factorial(value))
+#3
+# val = int(input("insert any int:" ))
+# while 0 < val < 100000:
+#     val=val*val
+#     print(val*val)
+#4
+# def Isprime(n):
```
Comment (itea-artem) [inline second_lesson_home_tasks.py]:
```suggestion
# def is_prime(n):
```

File: second_lesson_home_tasks.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+#         print('NO it is not prime number')
+#     else:
+#         print('YES it is prime number')
+#
+#
+# n = int(input("insert number: "))
+# Isprime(n)
+
+#5
+# summ = 0
```
Comment (itea-artem) [inline second_lesson_home_tasks.py]:
```suggestion
# sum = 0
```

File: second_lesson_home_tasks.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+# n = int(input("insert number: "))
+# if n < 1:
+#     print("value < 1 could not be calculated: ")
+# else:
+#     for i in range(1,(n+1)):
```
Comment (itea-artem) [inline second_lesson_home_tasks.py]:
```suggestion
#     for i in range(1, n+1):
```

File: second_lesson_home_tasks.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+#         if n % i == 0:
+#             summ += i
+#     print("sum of all devidors: ", summ)
+
+#6
+# n = int(input("insert number of lines: "))
+# for i in range(1,(n+1)):
```
Comment (itea-artem) [inline second_lesson_home_tasks.py]: PEP8
```suggestion
# for i in range(1, n+1):
```

File: second_lesson_home_tasks.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+#     cstr = i * "/\\"
+#     print(cstr.center(n*2, " "))
+
+#7
+# a, b = list(map(int, input("Insert two values separated by comma: ").split(',')))
```
Comment (itea-artem) [inline second_lesson_home_tasks.py]:
```suggestion
# a, b = map(int, input("Insert two values separated by comma: ").split(','))
```

File: second_lesson_home_tasks.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+# if a < b:
+#     print(a,b)
```
Comment (itea-artem) [inline second_lesson_home_tasks.py]: PEP8

File: second_lesson_home_tasks.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+#     ran = random.randint(int(a),int(b))
```
Comment (itea-artem) [inline second_lesson_home_tasks.py]: remove ints

File: second_lesson_home_tasks.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+#     guess = int(input("please try to guess value set: "))
+#     for i in range(a,b):
+#         if ran == guess:
+#             print("Ihhaaa, you're right")
+#             break
+#         elif ran > guess:
+#             guess = int(input("your value is lower then random one, try again: "))
+#             continue
```
Comment (itea-artem) [inline second_lesson_home_tasks.py]: no need for continue here

File: second_lesson_home_tasks.py [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+#         else:
+#             guess = int(input("your value is greater then random one, try again: "))
+#             continue
```
Comment (itea-artem) [inline second_lesson_home_tasks.py]: no need for continue here

File: second_lesson_tasks.py [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+import random
+import time
+'''tasks in lab, Constructions'''
+
+#1
+# for i in range(1, 11):
+#     print(i*i)
+
+#2
+# a=0
```
Comment (itea-artem) [inline second_lesson_tasks.py]: PEP8

File: second_lesson_tasks.py [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+# while a<=1000:
```
Comment (itea-artem) [inline second_lesson_tasks.py]: Wrong.

File: second_lesson_tasks.py [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+# a=input("put the value:" ) #to make one and more itterations running, put the value that is more than 100
+# while math.sqrt(int(a))>10:
+#     print(math.sqrt(int(a)))
+#     a=int(a)
+#     a=a-1
+
+#4
+# a="This is Hard timE fOr ouR countrY"
+# for i in a:
+#     if i.isupper() is True:
```
Comment (itea-artem) [inline second_lesson_tasks.py]:
```suggestion
#     if i.isupper():
```

**Review by itea-artem — CHANGES_REQUESTED** [review]

File: second_lesson_home_tasks.py (hunk 9 above)
Comment (ezjust) [reply second_lesson_home_tasks.py]: removed

File: second_lesson_home_tasks.py (hunk 10 above)
Comment (ezjust) [reply second_lesson_home_tasks.py]: removed

**Review by ezjust — COMMENTED** [review]
fixed all mistakes marked in scripts

**Comment by ezjust** [comment]
updated, fixed all mistakes, please recheck

File: second_lesson_home_tasks.py [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+    print('factorial is: ', math.factorial(value))
+#3
+value = int(input('insert any int:'))
+while 0 < value < 100000:
+    value = value*value
+    print(value*value)
+#4
+def is_prime(num):
+    if (math.factorial(num - 1) + 1) % num != 0:
+        print('NO it is not prime number')
```
Comment (itea-artem) [inline second_lesson_home_tasks.py]: return

File: second_lesson_home_tasks.py [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+            break
+        elif ran > guess:
+            guess = int(input('your value is lower then random one, try again: '))
+        else:
+            guess = int(input('your value is greater then random one, try again: '))
+elif a > b:
+    print('You have inserted bad values: {} > {} . Retry with integers'.format(a, b,))
+else:
+    print('You have inserted bad values: {} = {} and {} = {} . Retry with integers'.format(a, type(a), b, type(b)))
+
```
Comment (itea-artem) [inline second_lesson_home_tasks.py]: Blank

File: second_lesson_tasks.py [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+a = input('put the value:')   # make one and more iterations running, put the value that is more than 100
+while math.sqrt(int(a)) > 10:
+    print(math.sqrt(int(a)))
+    a = int(a)
+    a = a-1
+
+#4
+a = 'This is Hard timE fOr ouR countrY'
+for i in a:
+    if i.isupper() is True:
```
Comment (itea-artem) [inline second_lesson_tasks.py]:
```suggestion
    if i.isupper()
```

File: second_lesson_tasks.py [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+    digit = x % 10
+    t = t + digit
+    x = x//10
+print('total sum of digits in int is:', t)
+
+
+#9
+array = []
+for value in range(0, 10):
+    if value == 0:
```
Comment (itea-artem) [inline second_lesson_tasks.py]: Unnecessary check

File: second_lesson_tasks.py [hunk 18]
```diff
@@ -0,0 +1,197 @@
...
+#3
+x = int(input('insert integer:'))
+print('cosinus of x is: ', math.cos(x), '\nsinus of x is: ', math.sin(x))
+
+
+'''Function tasks'''
+#1
+
+
+def average_value(list):
```
Comment (itea-artem) [inline second_lesson_tasks.py]:
```suggestion
def get_average_value(list):
```

File: second_lesson_tasks.py [hunk 19]
```diff
@@ -0,0 +1,197 @@
...
+def sum_digits(list_1):
+    list2 = []
+    for i in list_1:
+        sum = (i % 10) + (i // 10)
+        list2.append(sum)
+    if len(list_1) < 2:
+        print('List was set with 1 element, so used default one, where sum of number equal: ')
+        sum_dig([33, 14, 15])
+    else:
+        return print(list2)
```
Comment (itea-artem) [inline second_lesson_tasks.py]:
```suggestion
        return list2
```

File: second_lesson_tasks.py [hunk 20]
```diff
@@ -0,0 +1,197 @@
...
+print("NO")
+
+#3
+list1 = []
+num = 3456789
+for digit in reversed(str(num)):
+    print(digit, end=" ")
+
+
+
```
Comment (itea-artem) [inline second_lesson_tasks.py]: blank

**Review by itea-artem — CHANGES_REQUESTED** [review]

File: second_lesson_tasks.py [hunk 21]
```diff
@@ -0,0 +1,188 @@
...
+    if number == 2**i:
+        print("YES")
+        exit()
+print("NO")
+
+#3
+list1 = []
+num = 3456789
+for digit in reversed(str(num)):
+    print(digit, end=" ")
```
Comment (itea-artem) [inline second_lesson_tasks.py]: New line

File: second_lesson_tasks.py [hunk 22]
```diff
@@ -0,0 +1,188 @@
...
+while a in range(0, 20):
+    a = a+1
+    if a % 2 == 0:
+        print(a)
+
+#7
+i = int(input('input any int: '))
+v_list = []
+while i != 0:
+    v_list.append(i)
```
Comment (itea-artem) [inline second_lesson_tasks.py]:
```suggestion
    i = int(input('input any int: '))
    v_list.append(i)
```

File: second_lesson_tasks.py [hunk 23]
```diff
@@ -0,0 +1,188 @@
...
+    i = int(input('input any int: '))
+    if i == 0:
```
Comment (itea-artem) [inline second_lesson_tasks.py]: remove this

File: second_lesson_tasks.py [hunk 24]
```diff
@@ -0,0 +1,188 @@
...
+        sum_dig([33, 14, 15])
+    else:
+        return list2
+
+
+sum_digits([13])
+
+#2
+number = int(input('insert number: '))
+for i in range(0, 64):
```
Comment (itea-artem) [inline second_lesson_tasks.py]: What if number == 2**100? This is recursion task

**Review by itea-artem — CHANGES_REQUESTED** [review]

File: second_lesson_tasks.py (hunk 21 above)
Comment (ezjust) [reply second_lesson_tasks.py]: didn't get what is the problem here?

File: check_if_value_simple.py [hunk 25]
```diff
@@ -0,0 +1,15 @@
+import math
+
+# Проверить является ли введенное число простым.
+# Число считается простым если оно не делится нацело на все числа до квадратного корня этого числа
+
+
+def is_prime(num):
```
Comment (itea-artem) [inline check_if_value_simple.py:7]: 2) Если функция возвращает True/False, она должна начинаться с is

File: check_power_of_two.py [hunk 26]
```diff
@@ -0,0 +1,11 @@
+# Дано натуральное число N. Выведите слово YES
+# если число N является точной степенью двойки, или слово NO в противном случае.
+
+number = int(input('insert number: '))
+for i in range(0, 64):
```
Comment (itea-artem) [inline check_power_of_two.py:5]: incorrect result for 2**65

File: cycle_power_integer.py [hunk 27]
```diff
@@ -0,0 +1,6 @@
+# Возвести число в степень с помощью цикла
+
+value = int(input('insert any int:'))
+while 0 < value < 100000:
+    value = value*value
+    print(value*value)
```
Comment (itea-artem) [inline cycle_power_integer.py:6]: Ваш код должен соблюдать pep8.

File: paint_triangle.py [hunk 28]
```diff
@@ -0,0 +1,6 @@
+# Нарисовать равнобедренный треугольник из символов ^. Высоту выбирает пользователь. Например: высота = 5, на экране
+
+num = int(input('insert number of lines: '))
+for i in range(1, (num + 1)):
+    line_pic = i * '/\\'
+    print(line_pic.center(num*2, ' '))
```
Comment (itea-artem) [inline paint_triangle.py:6]: Ваш код должен соблюдать pep8.

File: recursion_function_sum_digits.py [hunk 29]
```diff
@@ -0,0 +1,15 @@
+# Написать рекурсивную функцию, которая возвращает сумму цифр числа
+
+def sum_digits(list_1):
+    list2 = []
+    for i in list_1:
+        summation = (i % 10) + (i // 10)
+        list2.append(summation)
+    if len(list_1) < 2:
+        print('List was set with 1 element, so used default one, where sum of number equal: ')
+        sum_digits([33, 14, 15])
```
Comment (itea-artem) [inline recursion_function_sum_digits.py:10]: hardcoded data?

**Review by itea-artem — CHANGES_REQUESTED** [review]
4/5.5

**Pull request merged**